openeo_processes_dask/process_implementations/cubes/load.py

```diff
@@ -0,0 +1,164 @@
+        else:
+            raise Exception(
+                f"stac-validator returned multiple items, not supported yet. {stac.message}"
+            )
+        return is_valid_stac, asset_type
+
+    is_valid_stac, asset_type = validate_stac(url)
+
+    if not is_valid_stac:
+        raise Exception("The provided link is not a valid STAC.")
+
+    if asset_type == "COLLECTION":
+        # If query parameters are passed, try to get the parent Catalog if possible/exists, to use the /search endpoint
+        if spatial_extent or temporal_extent or bands or properties:
+            # If query parameters are passed, try to get the parent Catalog if possible/exists, to use the /search endpoint
+            # url = "https://earth-search.aws.element84.com/v0/collections/sentinel-s2-l2a-cogs"
+            parsed_url = urlparse(url)
+            root_url = parsed_url.scheme + "://" + parsed_url.netloc
+            catalog_url = root_url
+            url_parts = PurePosixPath(unquote(parsed_url.path)).parts
+            collection_id = url_parts[-1]
+            for p in url_parts:
+                if p != "/":
+                    catalog_url = catalog_url + "/" + p
+                try:
+                    is_valid_stac, asset_type = validate_stac(catalog_url)
+                except:
+                    continue
+                if is_valid_stac and asset_type == "CATALOG":
+                    break
+            if not is_valid_stac:
+                raise Exception(
+                    "It was not possible to find the root STAC Catalog starting from the provided Collection."
+                )
+
+            ## Check if we are connecting to Microsoft Planetary Computer, where we need to sign the connection
+            modifier = pc.sign_inplace if "planetarycomputer" in catalog_url else None
+
+            catalog = pystac_client.Client.open(catalog_url, modifier=modifier)
+
+            query_params = {"collections": [collection_id]}
+
+            if spatial_extent is not None:
+                try:
+                    spatial_extent_4326 = spatial_extent
+                    if spatial_extent.crs is not None:
+                        if not pyproj.crs.CRS(spatial_extent.crs).equals("EPSG:4326"):
+                            spatial_extent_4326 = _reproject_bbox(
+                                spatial_extent, "EPSG:4326"
+                            )
+                    bbox = [
+                        spatial_extent_4326.west,
+                        spatial_extent_4326.south,
+                        spatial_extent_4326.east,
+                        spatial_extent_4326.north,
+                    ]
+                    query_params["bbox"] = bbox
+                except Exception as e:
+                    raise Exception(f"Unable to parse the provided spatial extent: {e}")
+
+            if temporal_extent is not None:
+                query_params["datetime"] = [
+                    str(temporal_extent[0].to_numpy()),
```

Comment:
`temporal_extent[0]` can be a `str` according to the type annotation: what is this `to_numpy()` and how is that supposed to work?